amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/HierarchicalShardSyncer.java

```diff
@@ -164,6 +165,19 @@ public synchronized void checkAndCreateLeaseForNewShards(@NonNull final ShardDet
         }
     }
 
+    public synchronized Lease createLeaseForChildShard(ChildShard childShard) throws InvalidStateException {
```

Comment:
I think there is not much benefit to reuse the code from newKCLLease because the input parameter for this method is ChildShard, not Shard. And the way to populate parentShardIds is different for ChildShard. Also we need to checkpoint with TrimHorizon for createLeaseForChildShard, but not for newKCLLease.